static/src/js/components/GridGroupedSinglePanel/__tests__/GridGroupedSinglePanel.test.js

```diff
@@ -0,0 +1,172 @@
+import React from 'react'
+import { render, screen } from '@testing-library/react'
+import { userEvent } from '@testing-library/user-event'
+import GridGroupedSinglePanel from '../GridGroupedSinglePanel'
+
+jest.mock('../../CustomWidgetWrapper/CustomWidgetWrapper')
+
+const setup = (overrideProps = {}) => {
+  const onChange = jest.fn()
+  const layoutGridSchema = {
+    'ui:group': 'Index Ranges',
+    'ui:group-description': true,
+    'ui:group-single-panel': true,
+    'ui:row': [
+      {
+        'ui:col': {
+          md: 12,
+          children: [{
+            'ui:row': [{
+              'ui:col': {
+                md: 12,
+                children: ['LatRange']
+              }
+            }]
+          }, {
+            'ui:row': [{
+              'ui:col': {
+                md: 12,
+                children: ['LonRange']
+              }
+            }]
+          }]
+        }
+      }
+    ]
+  }
+
+  const uiSchema = {
+    'ui:field': 'layout',
+    'ui:title': 'Index Ranges',
+    'ui:layout_grid': {
+      'ui:group': 'Index Ranges',
+      'ui:group-description': true,
+      'ui:group-single-panel': true,
+      'ui:row': [{
+        'ui:col': {
+          md: 12,
+          children: [{
+            'ui:row': [{
+              'ui:col': {
+                md: 12,
+                children: ['LatRange']
+              }
+            }]
+          }, {
+            'ui:row': [{
+              'ui:col': {
+                md: 12,
+                children: ['LonRange']
+              }
+            }]
+          }]
+        }
+      }]
+    },
+    LatRange: {
+      'ui:canAdd': false,
+      items: { 'ui:type': 'number' }
+    },
+    LonRange: {
+      'ui:canAdd': false,
+      items: { 'ui:type': 'number' }
+    }
+  }
+
+  const schema = {
+    type: 'object',
+    additionalProperties: false,
+    description: 'This element describes the x and y dimension ranges for this variable. Typically these values are 2 latitude and longitude ranges, but they dont necessarily have to be.',
+    properties: {
+      LatRange: {
+        description: 'The LatRange consists of an index range for latitude.',
+        type: 'array',
+        items: { type: 'number' },
+        minItems: 2,
+        maxItems: 2
+      },
+      LonRange: {
+        description: 'The LonRange consists of an index range for longitude.',
+        type: 'array',
+        items: { type: 'number' },
+        minItems: 2,
+        maxItems: 2
+      }
+    },
+    required: ['LatRange', 'LonRange']
+  }
+
+  const props = {
+    layoutGridSchema,
```

Comment:
you canp probably just say: layout: uiSchema['ui:layout_grid'] and then remove the layout variable.